Techtonica/techtonica.org
# Job posting & payment feature

Reference materials:
https://developer.squareup.com/docs/sdks/python/quick-start
https://github.com/square/connect-api-examples/tree/master/connect-examples/v2/python_payment
https://developer.squareup.com/docs/devtools/sandbox/overview
https://developer.squareup.com/docs/devtools/sandbox/payments

---
### Initial Set Up Instructions

1. In the root directory, create a file `config.ini` containing this:
```
[default]
# Acceptable values are sandbox or production
environment = sandbox
dev_password = dev_password

[production]
square_application_id = production_application_id
square_access_token = production_access_token
square_location_id = production_location_id

[sandbox]
square_application_id = <sandbox app id>
square_access_token = <sandbox access token>
square_location_id = <sandbox location id>

[slack]
slack_webhook =  <slack webhook>
```
2. Fill in the `[sandbox]` values. You will need to log into **developer.squareup.com** and create an application (if you haven't already) in order to get the necessary tokens and ids. From there, you can access the dashboard and the API explorer to test payment routes. You'll also need to fill in the `slack_webhook` value with the correct key as well.


---
### Testing

1. Go to `<base url>/payment-form`
2. Fill out the form inputs
3. Use this card test input for a "successful transaction"
```
    card number: 4111 1111 1111 1111
    exp date: 12/24
    cvv: 123
    zip code: 94122
```
4. Submit
- [ ] You should see a "Payment successful!" message appear below
- [ ] On the Square developer dashboard, check the api logs and you should see a successful transaction with a matching timestamp
- [ ] In the appropriate slack channel, you should see a message sharing the values you input with a matching timestamp
---

**Remaining steps**

- [ ] Updating README documentation
- [ ] Adding additional field to job posting form
- [ ] Price selection(?)
- [ ] Adding website copy with user instructions
- [ ] Link to form on website
- [ ] Adding a frontend success message for successful slack notification
